plugin/keyring_vault/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+INCLUDE(CheckFunctionExists)
+INCLUDE(curl)
+
+SET(CMAKE_EXTRA_INCLUDE_FILES string.h)
+CHECK_FUNCTION_EXISTS(memset_s HAVE_MEMSET_S)
+SET(CMAKE_EXTRA_INCLUDE_FILES)
+
+MACRO (CHECK_IF_LIB_FOUND lib_name project_name)
+  SET (lib_found_variable_name "${lib_name}_FOUND")
+  IF (NOT DEFINED ${lib_found_variable_name} OR NOT ${${lib_found_variable_name}})
+    message("Not building ${project_name}, could not find library: ${lib_name}")
+    RETURN()
+  ENDIF()
+ENDMACRO()
+
+FIND_CURL()
+CHECK_IF_LIB_FOUND(CURL "keyring_vault")
+
+INCLUDE_DIRECTORIES(${BOOST_PATCHES_DIR})
+INCLUDE_DIRECTORIES(SYSTEM ${BOOST_INCLUDE_DIR})
+
+INCLUDE_DIRECTORIES(
+  ${CMAKE_SOURCE_DIR}/plugin/keyring/common
+  ${CURL_INCLUDE_DIRS})
```

Comment:
I am using FindCurl here, which comes with cmake. It produces the include dir as CURL_INCLUDE_DIRS.